server/accounts.go

```diff
@@ -4223,24 +4233,39 @@ func (s *Server) fetch(res AccountResolver, name string, timeout time.Duration)
 	respC := make(chan []byte, 1)
 	accountLookupRequest := fmt.Sprintf(accLookupReqSubj, name)
 	s.mu.Lock()
+	// Resolver will wait for available routes + gateways to reply
+	// before serving an error in case there weren't any found.
+	expectedServers := len(s.routes)
+	if s.gateway != nil {
```

Comment:
I think this is always non nil IIRC.

Reply:
Yes, instead the test could be:
```
if s.gateway.enabled {
..
```
There is currently no API to get the number of remotes, you could add one, but otherwise you should lock:
```
    s.gateway.RLock()
    expectedServers += len(s.gateway.remotes)
    s.gateway.RUnlock()
```

Reply:
Could use ``` s.sys.servers```

Reply:
Switched to use `s.sys.servers` to use the detected active servers instead.